projects/igniteui-angular/src/lib/grids/row-drag.directive.ts

```diff
@@ -0,0 +1,137 @@
+    public startDrag(event) {
+        this.onPointerDown(event);
+    }
+
+    @Input('igxRowDrag')
+    set data(val) {
+        this.row = val;
+    }
+
+    get data() {
+        return this.row;
+    }
+
+    public onPointerDown(event) {
+        event.preventDefault();
+        this._rowDragStarted = false;
+        super.onPointerDown(event);
+    }
+
+    public onPointerMove(event) {
+        super.onPointerMove(event);
+        if (this._dragStarted && !this._rowDragStarted) {
+            this._rowDragStarted = true;
+            const args: IRowDragStartEventArgs = {
+                owner: this,
+                dragData: this.row,
+                cancel: false
+            };
+
+            this.row.grid.onRowDragStart.emit(args);
+            if (args.cancel) {
+                this._clicked = false;
+                return;
+            }
+            this.row.dragging = true;
+            this.row.grid.rowDragging = true;
+            if (this.row.grid.rowEditable && this.row.grid.rowInEditMode) {
+                this.row.grid.endEdit(true);
+            }
+            this.row.grid.markForCheck();
+
+            this.subscription$ = fromEvent(this.row.grid.document.defaultView, 'keydown').subscribe((ev: KeyboardEvent) => {
+                if (ev.key === KEYS.ESCAPE || ev.key === KEYS.ESCAPE_IE) {
+                    this._lastDropArea = false;
+                    this.onPointerUp(event);
+                }
+            });
+        }
+    }
+
+    public onPointerUp(event) {
+        if (!this._clicked) {
+            return;
+        }
+        super.onPointerUp(event);
+        this.row.dragging = false;
+        this.row.grid.rowDragging = false;
+
+        const args: IRowDragEndEventArgs = {
+            owner: this,
+            dragData: this.row
+        };
+        this.zone.run(() => {
+            this.row.grid.onRowDragEnd.emit(args);
+        });
+        this._unsubscribe();
+    }
+
+    protected createDragGhost(event) {
+        super.createDragGhost(event, this.row.nativeElement);
+
+        const ghost = this._dragGhost;
+
+        const gridRect = this.row.grid.nativeElement.getBoundingClientRect();
+        const rowRect = this.row.nativeElement.getBoundingClientRect();
+        ghost.style.overflow = 'hidden';
+        ghost.style.width = gridRect.width + 'px';
+        ghost.style.height = rowRect.height + 'px';
+
+        ghost.classList = [];
+        this.renderer.addClass(ghost, this.row.defaultCssClass);
+        this.renderer.addClass(ghost, ghostBackgroundClass);
+
+        const ghostCells = ghost.getElementsByClassName(gridCellClass);
+        for (let index = 0; index < ghostCells.length; index++) {
+            ghostCells[index].classList = [gridCellClass];
+        }
+    }
+
+    private _unsubscribe() {
+        if (this.subscription$ && !this.subscription$.closed) {
+            this.subscription$.unsubscribe();
+            this.subscription$ = null;
```

Comment:
Do we need it set to null, or we can just delete subscription$?

Reply:
fixed